src/System.Private.Xml/src/System/Xml/Serialization/ReflectionXmlSerializationReader.cs

```diff
@@ -614,10 +616,100 @@ private static void AddObjectsIntoTargetCollection(object targetCollection, List
             }
         }
 
-        private static void SetMemberValue(object o, object value, string memberName)
+        private static IDictionary<Tuple<Type, Type>, Func<MemberInfo, SetMemberValueDelegate>> s_GetSetMemberValueDelegateWithTypeCache = new ConcurrentDictionary<Tuple<Type, Type>, Func<MemberInfo, SetMemberValueDelegate>>();
```

Comment:
Good point. I doubt it would make much difference as this shouldn't be on the hot code path, but it's good practice when trying to improve performance anyway.

---
In reply to: [122563884](https://github.com/dotnet/corefx/pull/21155#discussion_r122563884) [](ancestors = 122563884)